packages/editor/package.json

```diff
@@ -62,6 +62,7 @@
 		"@wordpress/private-apis": "file:../private-apis",
 		"@wordpress/reusable-blocks": "file:../reusable-blocks",
 		"@wordpress/rich-text": "file:../rich-text",
+		"@wordpress/router": "file:../router",
```

Comment:
This is the major concern I have to land this PR.

By doing this, the `@wordpress/editor` package would depend on some utilities provided by the router that require a `<RouterProvider>` in the component tree. Is this something the editor package cares about? Or is it the higher-up edit-site/edit-post/etc. responsibility instead? @youknowriad I presume you have thoughts about this and can share advice.

Reply:
This is a bit of a gray area.

If I understand properly, you want to move fields to "editor" (or core-data) or something like that. That's totally fine by me assuming all the fields are independent of the place they are used in (post editor, site editor...). 

The problem is that the urls them selves are specific, in the site editor you want to navigate to some places but not in the post editor... Maybe for a start we can use always the same url and it should work regardless of where you are.

The other problem is that I'm not sure "urls" make sense in fields at all. I think for instance rendering "Media" or "Title" field or something shouldn't actually render an application url/link automatically. That feels like a separate config that should be passed to `DataViews`. For instance a `getItemLink` just like we have `getItemId`.

Also nothing that we already have something like `getItemLabel` but it's not implemented properly, right now we guess it by fetching the "primary" field which IMO is mixing separate concerns.

Reply:
> If I understand properly, you want to move fields to "editor" (or core-data) or something like that.

Yeah. One of the issues I want to fix is: how do we share the field definition among all places where it's used? They need to be available in three places now: edit-site/post-edit, edit-site/post-list, editor/post-actions.

Reply:
Yes, just like the actions. 

My goal for the actions was to move them one by one (and type them) into `editor/src/dataviews/actions` and I can see the same for fields in `editor/src/dataviews/fields`.

I think ultimately, it makes sense for `editor/src/dataviews` to be its own package: `core-dataviews` or something or actually be part of `core-data`